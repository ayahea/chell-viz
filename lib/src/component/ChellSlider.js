"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var rc_slider_1 = require("rc-slider");
var React = require("react");
require("rc-slider/assets/index.css");
/**
 * Represents a simple 2d slider, allowing a value to be selected between a minimum and maximum.
 *
 * @export
 * @extends {React.Component<IChellSliderProps, IChellSliderState>}
 */
var ChellSlider = /** @class */ (function (_super) {
    __extends(ChellSlider, _super);
    function ChellSlider(props) {
        var _this = _super.call(this, props) || this;
        /**
         * Updates the state of the slider after user commits to change. If applicable, invokes appropriate callback.
         */
        _this.onAfterChange = function (cb) { return function (value) {
            if (cb) {
                cb(value);
            }
            _this.setState({
                value: value,
            });
        }; };
        /**
         * Updates the state of the slider as the user moves the slider around but before selection is committed.
         * If applicable, invokes appropriate callback as well.
         */
        _this.onChange = function (cb) { return function (value) {
            if (cb) {
                cb(value);
            }
            _this.setState({
                value: value,
            });
        }; };
        _this.state = {
            value: props.defaultValue,
        };
        return _this;
    }
    ChellSlider.prototype.render = function () {
        var _a = this.props, defaultValue = _a.defaultValue, max = _a.max, min = _a.min, label = _a.label, onAfterChange = _a.onAfterChange, onChange = _a.onChange, style = _a.style;
        return (React.createElement("div", null,
            React.createElement("p", null, label + ": " + this.state.value),
            React.createElement(rc_slider_1.default, { defaultValue: defaultValue, max: max, min: min, onAfterChange: this.onAfterChange(onAfterChange), onChange: this.onChange(onChange), style: style })));
    };
    return ChellSlider;
}(React.Component));
exports.ChellSlider = ChellSlider;
