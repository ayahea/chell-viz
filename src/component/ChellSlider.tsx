import Slider from 'rc-slider';
import * as React from 'react';

import 'rc-slider/assets/index.css';

/** Function signature that is invoked on slider events. */
export type ChellSliderCallback =
  /**
   * @param {number} value New value the slider is currently on top of.
   */
  (value: number) => void;

/**
 * Properties that ChellSlider accepts.
 *
 * @export
 * @interface IChellSliderProps
 */
export interface IChellSliderProps {
  /** Initial value the slider is set to. */
  defaultValue: number;

  /** Initial value the slider is set to. */
  label: string;

  /** Maximum value for slider. */
  max: number;

  /** Minimum value for slider. */
  min: number;

  /** Invoked when the value has finished changing the slider value - usually by releasing the mouse. */
  onAfterChange?: ChellSliderCallback;

  /** Invoked when the value is in the middle of changing but user has not committed to the change. */
  onChange?: ChellSliderCallback;
}

/**
 * State of the Chell Slider.
 *
 * @export
 * @interface IChellSliderState
 */
export interface IChellSliderState {
  /** Value the slider is currently set to. */
  value: number;
}

/**
 * Represents a simple 2d slider, allowing a value to be selected between a minimum and maximum.
 *
 * @export
 * @class ChellSlider
 * @extends {React.Component<IChellSliderProps, IChellSliderState>}
 */
export class ChellSlider extends React.Component<IChellSliderProps, IChellSliderState> {
  constructor(props: IChellSliderProps) {
    super(props);
    this.state = {
      value: props.defaultValue,
    };
  }

  public render() {
    const { defaultValue, max, min, label, onAfterChange, onChange } = this.props;
    return (
      <div>
        <p>{`${label}: ${this.state.value}`}</p>
        <Slider
          defaultValue={defaultValue}
          max={max}
          min={min}
          onAfterChange={this.onAfterChange(onAfterChange)}
          onChange={this.onChange(onChange)}
        />
      </div>
    );
  }

  /**
   * Updates the state of the slider after user commits to change. If applicable, invokes appropriate callback.
   *
   * @private
   * @memberof ChellSlider
   */
  private onAfterChange = (cb?: ChellSliderCallback) => (value: number) => {
    if (cb) {
      cb(value);
    }
    this.setState({
      value,
    });
  };

  /**
   * Updates the state of the slider as the user moves the slider around but before selection is committed.
   * If applicable, invokes appropriate callback as well.
   *
   * @private
   * @memberof ChellSlider
   */
  private onChange = (cb?: ChellSliderCallback) => (value: number) => {
    if (cb) {
      cb(value);
    }
    this.setState({
      value,
    });
  };
}
