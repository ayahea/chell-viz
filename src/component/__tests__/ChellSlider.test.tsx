import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { ChellSlider } from '../ChellSlider';

describe('ChellSlider', () => {
  test('Should match existing snapshot when given simple data.', () => {
    const chellSliderWrapper = shallow(<ChellSlider defaultValue={10} max={100} min={1} label={'test'} />);
    expect(toJson(chellSliderWrapper)).toMatchSnapshot();
  });
});
