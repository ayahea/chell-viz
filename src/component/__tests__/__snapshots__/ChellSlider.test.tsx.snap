// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChellSlider Should match existing snapshot when given simple data. 1`] = `
<div>
  <p>
    test: 10
  </p>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className=""
    defaultValue={10}
    disabled={false}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={true}
    marks={Object {}}
    max={100}
    min={1}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={Object {}}
    step={1}
    trackStyle={
      Array [
        Object {},
      ]
    }
    vertical={false}
  />
</div>
`;
